presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/MetadataResolver.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.analyzer;
+
+import com.facebook.presto.common.CatalogSchemaName;
+import com.facebook.presto.common.QualifiedObjectName;
+import com.facebook.presto.common.ViewDefinition;
+import com.facebook.presto.common.predicate.TupleDomain;
+import com.facebook.presto.common.type.ParametricType;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.spi.ColumnMetadata;
+import com.facebook.presto.spi.MaterializedViewDefinition;
+import com.facebook.presto.spi.MaterializedViewStatus;
+import com.facebook.presto.spi.function.SqlFunction;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Metadata resolver provides information about Catalog, Schema, Tables, Views, Types, and Functions required for analyzer functionality.
+ */
+public interface MetadataResolver
```

Comment:
The design of the class is to the wrong direction? I would expect the implementation of this interface to be engines (in this case, Presto) instead of analyzer? Meaning, analyzer should call these methods to retrieve metadata. But the implementation indicates it is  part of analyzer?

Reply:
We discussed on this offline, we will revisit these changes in a separate PR.